File: FMAplication/Domain/Campaign/Campaign.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using FMAplication.Core;

namespace FMAplication.Domain.Campaign
{
    public class Campaign: AuditableEntity<int>
    {
        //public int CampaignId { get; set; }   
        [StringLength(255)]
        public string CampaignName { get; set; }  
        [StringLength(Int32.MaxValue)]
        public string CampaignDetails { get; set; }

        public DateTime StartDate { get; set; }            
        public DateTime EndDate { get; set; }

        public IList<CampaignHistory> CampaignHistories { get; set; }
    }
}

File: FMAplication/ClientApp/src/app/Shared/Entity/Campaigns/campaign.ts
import { Status } from "src/app/Shared/Enums/status";
import { CampaignHistory } from './campaign-history';

export class Campaign {
    public id: number;
    public campaignName: string;
    public campaignDetails: string;
    public startDate: Date;
    public endDate: Date;
    public status: number;

    public campaignHistories: CampaignHistory[];

    constructor() {
        this.id = 0;
        this.campaignName = '';
        this.campaignDetails = '';
        this.status = Status.Active;

        this.campaignHistories = [];
    }
}